pkg/skaffold/jib/jib.go

```diff
@@ -78,3 +100,33 @@ func getDependencies(cmd *exec.Cmd) ([]string, error) {
 	sort.Strings(deps)
 	return deps, nil
 }
+
+// calculateRoots returns a list of possible symlink-expanded paths
+func calculateRoots(path string) ([]string, error) {
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to resolve %s", path)
+	}
+	canonical, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to canonicalize workspace %s", path)
+	}
+	if path == canonical {
+		return []string{path}, nil
+	}
+	return []string{canonical, path}, nil
+}
+
+// relativize tries to make path relative to one of the given roots
+func relativize(path string, roots ...string) (string, error) {
+	if !filepath.IsAbs(path) {
+		return path, nil
+	}
+	for _, root := range roots {
+		// check that the path can be made relative and is contained (since `filepath.Rel("/a", "/b") => "../b"`)
+		if rel, err := filepath.Rel(root, path); err == nil && !strings.HasPrefix(rel, dotDotSlash) {
```

Comment:
I think we should handle the errors here explicitly as well - it should be easier to debug errors.

Reply:
Maybe concatenate them, if there are more. There are max two roots as I'm reading this...

Reply:
These aren't real errors: it just means that the file could not be made relative, which usually means that `path` is already relative.  (Can also happen on Windows if they're on different volumes.)